### ISSUE TITLE
Abstraction layer for SensorThings API query and filter functions

### ISSUE THREAD

**Issue by timEngbrocks** [FraunhoferIOSB/FROST-Client#52]
# SensorThings API: query and filter functions
The SensorThings API offers many built-in query and filter functions for queries to the SensorThings Service.
Some examples:
- st_within: Find things in an area
- eq: Find things where some property equals some value
- distance: Calculate the distance between two points

The full documentation can be found here: [OGC SensorThings API Part 1: Sensing](https://docs.opengeospatial.org/is/15-078r6/15-078r6.html#56) in section 9.3.3.5

# Current state
Currently it is possible to add filters to queries by calling filter() on a query: [method definition](https://github.com/FraunhoferIOSB/FROST-Client/blob/master/src/main/java/de/fraunhofer/iosb/ilt/sta/query/Query.java#L80)
This method takes a string which is placed into the query which means that part of the query has to be defined manually.

# Suggestion
It would be great if filters could be controlled with built-in methods.
A query could for example have a list of filters where each filter has a operation and a list of arguments.
This would provide a layer of abstraction so that users of the FROST-Client don't have to know the SensorThings API Syntax.

**Comment by hylkevds** [comment]
Yes, this would be very nice to have, but also a lot of work to implement.

**Comment by hylkevds** [comment]
This is also related to #37, since Filters can be used in Expands...
